libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,164 @@
+// (C) 2022 GoodData Corporation
+import React from "react";
+import cx from "classnames";
+import { DateRangePickerInputField } from "./DateRangePickerInputField";
+import { injectIntl, WrappedComponentProps } from "react-intl";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import DayPickerInput from "react-day-picker/DayPickerInput";
+import moment from "moment";
+import { DateRangePickerInputFieldBody } from "./DateRangePickerInputFieldBody";
+import { convertPlatformDateStringToDate } from "../utils/DateConversions";
+import { TIME_FORMAT } from "../constants/Platform";
+
+interface IDateTimePickerOwnProps {
+    placeholderDate: string;
+    dateFormat: string;
+    onChange: (value: Date) => any;
+    value: Date;
+    dayPickerPropsWithDefaults: any;
+    handleDayClick: any;
+    locale: string;
+    isMobile: boolean;
+    isTimeForStaticPeriodEnabled: boolean;
+    className: string;
+    defaultTime?: string;
+    error?: boolean;
+}
+
+export type DateTimePickerComponentProps = IDateTimePickerOwnProps & WrappedComponentProps;
+
+// eslint-disable-next-line react/display-name
+export const DateTimePickerComponent = React.forwardRef<DayPickerInput, DateTimePickerComponentProps>(
+    (props: DateTimePickerComponentProps, ref) => {
+        const {
+            placeholderDate,
+            value,
+            onChange,
+            dateFormat,
+            dayPickerPropsWithDefaults,
+            handleDayClick,
+            isMobile,
+            isTimeForStaticPeriodEnabled,
+            className,
+            defaultTime = "00:00",
+            error = false,
+        } = props;
+
+        const getFormattedDate = () => {
+            return moment(value).format("YYYY-MM-DD");
+        };
+
+        const getFormattedTime = () => {
+            if (typeof value !== ("undefined" || null)) {
+                return moment(value).format(TIME_FORMAT);
+            }
+            return moment(null).format(TIME_FORMAT);
+        };
+
+        const onDateChange = (selectedDate: Date) => {
+            if (isTimeForStaticPeriodEnabled && selectedDate) {
+                let oldDate = value;
+                if (!oldDate) {
+                    oldDate = moment(defaultTime).toDate();
+                }
+
+                let updatedDatetime = moment(selectedDate).hours(oldDate.getHours());
+                updatedDatetime = moment(updatedDatetime).minutes(oldDate.getMinutes());
+
+                // set default time if necessary && configured
+                if (updatedDatetime.isSame(selectedDate)) {
+                    const time = moment(defaultTime, TIME_FORMAT).toDate();
+                    updatedDatetime.hours(time.getHours());
+                    updatedDatetime.minutes(time.getMinutes());
+                }
+
+                selectedDate = updatedDatetime.toDate();
```

Comment:
It is against good practice to modify function parameters. Either by changing the object value or reassign it.

I would move the whole content of the `if` branch into the function outside of the component and call it based on what it does (set time if not set? I'm not sure 100%). Function should start with the `if` condition that determines if whole operation is necessary and if not, return `selectedDate` otherwise it will continue and return new date setup as necessary. This way you don't need to override params of function and have the function more clear. Currently, it is not easy for me to understand why it does what it do now so try to make it more clear, please.

Reply:
Good point. I moved it into `adjustDate` method.